interop-cats/jvm/src/main/scala/scalaz/zio/interop/Schedule.scala

```diff
@@ -0,0 +1,598 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio.interop
+
+import cats.effect.ConcurrentEffect
+import scalaz.zio.ZSchedule.Decision
+import scalaz.zio.clock.Clock
+import scalaz.zio.{ ZSchedule, _ }
+import scalaz.zio.duration.{ Duration => ZDuration }
+import scalaz.zio.random.Random
+
+import scala.concurrent.duration.Duration
+
+/**
+ * See [[scalaz.zio.ZSchedule]]
+ */
+trait Schedule[F[+ _], -A, +B] { self =>
+  import Schedule.{ toEffect, Env }
+
+  implicit def F: ConcurrentEffect[F]
```

Comment:
Instead of placing this here, consider just requiring it (or something less restrictive, if possible) at individual method sites.

In addition, I'd make `Schedule` a class with the single `underlying` field.

You can see an example of how this is being done with `STM`.